components/forms/Form/Form.tsx

```diff
@@ -19,15 +25,50 @@ declare global {
  * @param props
  */
 const InnerForm = (props: InnerFormProps & FormikProps<FormValues> & DynamicFormProps) => {
-  const { children, handleSubmit, t, isSubmitting } = props;
+  const { children, handleSubmit, t, isSubmitting, isReCaptchaEnableOnSite } = props;
   const [canFocusOnError, setCanFocusOnError] = useState(false);
   const [lastSubmitCount, setLastSubmitCount] = useState(0);
-
   const errorList = props.errors ? getErrorList(props) : null;
   const errorId = "gc-form-errors";
   const serverErrorId = `${errorId}-server`;
   const formStatusError = props.status === "Error" ? t("server-error") : null;
 
+  const handleFormSubmission = (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    window.grecaptcha.ready(() => {
+      // get reCAPTCHA response
+      window.grecaptcha
+        .execute(process.env.NEXT_PUBLIC_RECAPTCHA_V3_SITE_KEY, { action: "submit" })
+        .then(async (clientToken: string) => {
+          sendClientTokenForVerification(clientToken)
+            .then((res) => {
+              const { score, success } = res.data;
+              logMessage.warn(`score : ${score}  status: ${success}`);
+              // assuming you're not a Robot and  submit
+              handleSubmit(evt);
+            })
+            .catch((error) => {
+              logMessage.error(error);
+            });
+        });
+    });
+  };
+
+  const sendClientTokenForVerification = async (token: string) => {
+    // call a backend API to verify reCAPTCHA response
+    return await axios({
```

Comment:
What is the reasoning behind making a request to our API in order to make a request to Google's API? It increases the chances of getting a failure.

Reply:
I guess first for Security reason, we need to access our API secret from the backend that way we avoid having the secret key in the FrondEnd.